CMakeLists.txt

```diff
@@ -139,12 +139,14 @@ list(APPEND DDPROF_INCLUDE_LIST ${CMAKE_SOURCE_DIR}/include ${CMAKE_SOURCE_DIR}/
 # Find the source files
 aux_source_directory(src COMMON_SRC)
 aux_source_directory(src/pprof PPROF_SRC)
+aux_source_directory(src/demangler DEMANGLER_SRC)
```

Comment:
Not sure a directory is need for a single file

Reply:
Yeah, I had mixed feelings about this.  I wanted to keep the implementation pure so that the Java profiler could use it as well, but I think they'll only care about C++ demangling anyway, so this implementation will be slightly unnecessary for them.

If it's OK with you, I'll ignore your feedback for the purpose of this PR, merge (when/if approved), propose to the Java team, and almost definitely spin out a subsequent PR where I do exactly what you suggest here and in the comment about the namespace.  :)